Source/fn_enforce_wall_bcs.cpp

```diff
@@ -273,3 +273,68 @@ void enforce_boundary_conditions (MultiFab& velCart,
 #endif
     }
 }
+
+void enforce_wall_bcs_for_face_centered_velocities_on_physical_boundaries (MultiFab& velCart,
+                                                                           Array<MultiFab, AMREX_SPACEDIM>& velCont,
+                                                                           Geometry const& geom)
+{
+    Box dom(geom.Domain());
+
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (Gpu::notInLaunchRegion())
+#endif
+
+    for ( MFIter mfi(velCont[0]); mfi.isValid(); ++mfi )
+    {
+        int const& box_id = mfi.LocalIndex();
+        amrex::Print() << "INFO| Box ID: " << box_id << "\n";
+
+        const Box& xbx = mfi.tilebox(IntVect(AMREX_D_DECL(1,0,0)));
+        const Box& ybx = mfi.tilebox(IntVect(AMREX_D_DECL(0,1,0)));
+#if (AMREX_SPACEDIM > 2)
+        const Box& zbx = mfi.tilebox(IntVect(AMREX_D_DECL(0,0,1)));
+#endif
+        auto const& vel_cont_x = velCont[0].array(mfi);
+        auto const& vel_cont_y = velCont[1].array(mfi);
+#if (AMREX_SPACEDIM > 2)
+        auto const& vel_cont_z = velCont[2].array(mfi);
+#endif
+
+        auto const& vel_cart = velCart.array(mfi);
+
+        int lo = dom.smallEnd(0);
+        int hi = dom.bigEnd(0)+1;
+
+        amrex::ParallelFor(xbx,
+                           [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+            if ( i == lo || i == hi ) {
```

Comment:
If your intention is this to be a wall, you need to pass "phy_bc_lo" into this function and check whether or not phy_bc_lo[0]==-1 or phy_bc_lo[1]==1 (i.e., it's a wall).

Also I would think that if it were a wall, the normal velocity should simply be set to zero.  What are your thoughts?

Reply:
I put the wall enforce function inside the cont2cart so "phy_bc_xx" has already been checked before that. Let me double-check!

Reply:
I don't see the logic in cont2cart checking for walls - you simply call this function.

Reply:
Oh yes, I checked it for the function that enforces the wall on cell-centered nodes but somehow not for this one. I will include the "phy_bc_xx" to the function.